docs/ingress-tls.md

```diff
@@ -0,0 +1,285 @@
+* Set secret type to `kubernetes.io/tls`
+
+```bash
+export TENANT_ID=[YOUR TENANT ID]
+
+cat <<EOF | kubectl apply -n $NAMESPACE -f -
+apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
+kind: SecretProviderClass
+metadata:
+  name: azure-tls
+spec:
+  provider: azure
+  secretObjects:                                # secretObjects defines the desired state of synced K8s secret objects
+  - secretName: ingress-tls-csi
+    type: kubernetes.io/tls
+    data: 
+    - objectName: $CERT_NAME
+      key: tls.key
+    - objectName: $CERT_NAME
+      key: tls.crt
+  parameters:
+    usePodIdentity: "false"
+    keyvaultName: $AKV_NAME                        # the name of the KeyVault
+    objects: |
+      array:
+        - |
+          objectName: $CERT_NAME
+          objectType: secret
+    tenantId: $TENANT_ID                  # the tenant ID of the KeyVault
+EOF
+```
+
+## Deploy Ingress Controller
+
+### Add the official ingress chart repository
+
+```bash
+helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx
+helm repo add stable https://kubernetes-charts.storage.googleapis.com/
+helm repo update
+```
+
+### Deploy Ingress
+
+Depending on the TLS certificate lifecycle, follow one of the following steps:
+
+* #### Bind certificate to Application
+
+> Application's deployment will reference the csi store provider.
+
+```bash
+helm install ingress-nginx/ingress-nginx --generate-name \
+    --namespace $NAMESPACE \
+    --set controller.replicaCount=2 \
+    --set controller.nodeSelector."beta\.kubernetes\.io/os"=linux \
+    --set defaultBackend.nodeSelector."beta\.kubernetes\.io/os"=linux
+```
+
+Next, [Deploy the application](#Deploy-Application-with-Reference-to-Secrets-Store-CSI).
+
+* #### Bind certificate to Ingress
+
+> NOTE: Ingress controller references a Secrets Store CSI volume and a `secretProviderClass` object created earlier. A Kubernetes secret `ingress-tls-csi` will be created by the CSI driver as a result of ingress creation.
+
+```bash
+helm install ingress-nginx/ingress-nginx --generate-name \
+    --namespace $NAMESPACE \
+    --set controller.replicaCount=2 \
+    --set controller.nodeSelector."beta\.kubernetes\.io/os"=linux \
+    --set defaultBackend.nodeSelector."beta\.kubernetes\.io/os"=linux \
+    --set controller.podLabels.aadpodidbinding=$AAD_POD_IDENTITY_NAME \
+    -f - <<EOF
+controller:
+  extraVolumes:
+      - name: secrets-store-inline
+        csi:
+            driver: secrets-store.csi.k8s.io
+            readOnly: true
+            volumeAttributes:
+              secretProviderClass: "azure-tls"
+            nodePublishSecretRef:
+              name: secrets-store-creds
+  extraVolumeMounts:
+      - name: secrets-store-inline
+        mountPath: "/mnt/secrets-store"
+        readOnly: true
+EOF
+```
+
+If not using [service principal mode](./service-principal-mode.md), remove the following snippet from the script:
+
+```bash
+--set controller.podLabels.aadpodidbinding=$AAD_POD_IDENTITY_NAME \
```

Comment:
this is still required if not using service principal mode

Reply:
fixed